lib/components/viewers/stop-time-cell.js

```diff
@@ -0,0 +1,146 @@
+
+/**
+ * If the browser timezone is not the homeTimezone, appends a time zone designation (e.g., PDT)
+ * to designate times in the homeTimezone
+ * (e.g., user in New York, but viewing a trip planner for service based in Los Angeles).
+ */
+function getTimeFormat (timeFormat, homeTimezone) {
+  const userTimeZone = getUserTimezone()
+  const inHomeTimezone = homeTimezone && homeTimezone === userTimeZone
+  return inHomeTimezone ? timeFormat : `${timeFormat} z`
+}
+
+/**
+ * Renders a stop time as either schedule or countdown, with an optional status icon.
+ * Stop time that apply to a different day have an additional text showing the day of departure.
+ */
+const StopTimeCell = ({
+  homeTimezone,
+  showIcon,
+  soonText,
+  stopTime,
+  timeFormat,
+  useCountdown,
+  useSchedule
+}) => {
+  const departureTime = useSchedule
+    ? stopTime.scheduledDeparture
+    : stopTime.realtimeDeparture
+  const now = moment().tz(homeTimezone)
+  const serviceDay = moment(stopTime.serviceDay * 1000).tz(homeTimezone)
+
+  // Determine if arrival occurs on different day, making sure to account for
+  // any extra days added to the service day if it arrives after midnight. Note:
+  // this can handle the rare (and non-existent?) case where an arrival occurs
+  // 48:00 hours (or more) from the start of the service day.
+  const departureTimeRemainder = departureTime % ONE_DAY_IN_SECONDS
+  const daysAfterServiceDay = (departureTime - departureTimeRemainder) / ONE_DAY_IN_SECONDS
+  const departureDay = serviceDay.add(daysAfterServiceDay, 'day')
+  const vehicleDepartsToday = now.dayOfYear() === departureDay.dayOfYear()
+
+  // Determine whether to show departure as countdown (e.g. "5 min") or as HH:mm
+  // time.
+  const secondsUntilDeparture = getSecondsUntilDeparture(stopTime, useSchedule)
+  // Determine if vehicle arrives after midnight in order to advance the day of
+  // the week when showing arrival time/day.
+  const departsInFuture = secondsUntilDeparture > 0
+  // Show the exact time if the departure happens within an hour.
+  const showCountdown = useCountdown && secondsUntilDeparture < ONE_HOUR_IN_SECONDS && departsInFuture
+
+  // Use "soon text" (e.g., Due) if vehicle is approaching.
+  const countdownString = secondsUntilDeparture < 60
+    ? soonText
+    : formatDuration(secondsUntilDeparture)
+  const formattedTime = formatSecondsAfterMidnight(
+    departureTime,
+    // Only show timezone (e.g., PDT) if user is not in home time zone (e.g., user
+    // in New York, but viewing a trip planner for service based in Los Angeles).
+    getTimeFormat(timeFormat, homeTimezone)
+  )
+  // We only want to show the day of the week if the arrival is on a
+  // different day and we're not showing the countdown string. This avoids
+  // cases such as when it's Wednesday at 11:55pm and an arrival occurs at
+  // Thursday 12:19am. We don't want the time to read: 'Thursday, 24 minutes'.
+  const showDayOfWeek = !vehicleDepartsToday && !showCountdown
```

Comment:
I don't think the day of the week should be shown in scheduledView.

Reply:
Addressed in https://github.com/opentripplanner/otp-react-redux/pull/292/commits/edd5a4bf88907df9d012180bedebaaddbc01d6f8.